### DISCUSSION TITLE
Default Assistants Configuration

### DISCUSSION THREAD

**Discussion by ev-caseythefirst** [danny-avila/LibreChat#2437] [Feature Requests & Suggestions]
### What happened?

This could be a two part-er.

**Part 1: librechat.yaml config**
While trying to configure the Assistants endpoint, I had removed all the custom endpoints and was focusing on the endpoints:assistants: section in the librechat.yaml file.  I was having a heck of a time disabling the assistants builder, even though it was a simple boolean value in the config.  Upon inspection of the debug and error logs, I realized that it was complaining about an invalid config file (even though a yaml validator validated it).  Turns out the empty custom: tag under the endpoints was throwing an error (invalid type: expecting array) so it wasn't loading the yaml file correctly (there was no ui error, only found in the log files).

The fix was to just remove the empty config: tag and then the yaml file was loading and acting as expected.

**Part 2: default assistant configuration**
Given that I was trying to disable the builder and restrict access to an individual assistant by id, the fact that the yaml file was error'ing out rather silently meant that ALL assistants on the account were exposed, and the builder was enabled.

I would think that explicitly enabling the builder through a valid config option would be more secure than explicitly disabling it.  Furthermore, perhaps making no assistants available upon an invalid config would be more secure functionality?

### Steps to Reproduce

Edit librechat.yaml file. 

delete everything in endpoints: custom:, leaving the custom tag.

### Relevant log output

```shell
{"issues":[{"code":"invalid_type","expected":"array","message":"Expected array, received null","path":["endpoints","custom"],"received":"null"}],"level":"error","message":"Invalid custom config file at /app/librechat.yaml [\n  {\n    \"code\": \"invalid_type\",\n    \"expected\": \"array\",\n    \"received\": \"null\",\n    \"path\": [\n      \"endpoints\",\n      \"custom\"\n    ],\n    \"message\": \"Expected array, received null\"\n  }\n]","name":"ZodError","stack":"ZodError: [\n  {\n    \"code\": \"invalid_type\",\n    \"expected\": \"array\",\n    \"received\": \"null\",\n    \"path\": [\n      \"endpoints\",\n      \"custom\"\n    ],\n    \"message\": \"Expected array, received null\"\n  }\n]\n    at get error [as error] (/app/node_modules/zod/lib/types.js:43:31)\n    at loadCustomConfig (/app/api/server/services/Config/loadCustomConfig.js:59:83)\n    at AppService (/app/api/server/services/AppService.js:35:25)\n    at startServer (/app/api/server/index.js:36:9)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
There are some options that throw an error and won’t let you start the server if the config file is invalid. Maybe I can add this to the list of errors that do so, or make them all throw an error
